### PR TITLE
Put back Travis, Make Tests Lighter

### PR DESCRIPTION
**Pull request by koaning**

### CODE REVIEW

**Comment by koaning** [comment]
I have also made some changes to the test scripts. Since the project is nearing 2000 tests. It would be good to keep an eye on how long everything runs. I've made some changes to the slowest tests. Here's the results from running `pytest --durations 0`. 

**before** 

```
29.91s call     tests/test_estimators/test_deadzone.py::test_deadzone[check_shape_remains_same_regressor]
15.70s call     tests/test_estimators/test_gmm_naive_bayes.py::test_estimator_checks[check_shape_remains_same_classifier]
12.57s call     tests/test_estimators/test_demographic_parity.py::test_same_logistic_multiclass[random_xy_dataset_multiclf25]
10.65s call     tests/test_estimators/test_equal_opportunity.py::test_same_logistic_multiclass[random_xy_dataset_multiclf25]
10.33s call     tests/test_estimators/test_equal_opportunity.py::test_same_logistic_multiclass[random_xy_dataset_multiclf26]
10.24s call     tests/test_estimators/test_demographic_parity.py::test_same_logistic_multiclass[random_xy_dataset_multiclf26]
8.04s call     tests/test_estimators/test_demographic_parity.py::test_same_logistic_multiclass[random_xy_dataset_multiclf24]
7.46s call     tests/test_estimators/test_equal_opportunity.py::test_same_logistic_multiclass[random_xy_dataset_multiclf24]
3.91s call     tests/test_estimators/test_demographic_parity.py::test_same_logistic[random_xy_dataset_clf26]
3.66s call     tests/test_estimators/test_demographic_parity.py::test_same_logistic[random_xy_dataset_clf25]
3.54s call     tests/test_estimators/test_equal_opportunity.py::test_same_logistic_multiclass[random_xy_dataset_multiclf21]
3.45s call     tests/test_estimators/test_equal_opportunity.py::test_same_logistic_multiclass[random_xy_dataset_multiclf22]
3.45s call     tests/test_estimators/test_equal_opportunity.py::test_same_logistic[random_xy_dataset_clf25]
3.44s call     tests/test_estimators/test_demographic_parity.py::test_same_logistic_multiclass[random_xy_dataset_multiclf21]
3.33s call     tests/test_estimators/test_equal_opportunity.py::test_same_logistic_multiclass[random_xy_dataset_multiclf23]
```

**after**
```
13.39s call     tests/test_estimators/test_deadzone.py::test_deadzone[check_shape_remains_same_regressor]
8.13s call     tests/test_estimators/test_gmm_naive_bayes.py::test_estimator_checks[check_shape_remains_same_classifier]
1.42s call     tests/test_estimators/test_gmm_naive_bayes.py::test_estimator_checks[check_estimators_dtypes]
1.17s call     tests/test_estimators/test_gmm_naive_bayes.py::test_estimator_checks[check_classifiers_train]
1.06s call     tests/test_estimators/test_equal_opportunity.py::test_same_logistic_multiclass[random_xy_dataset_multiclf26]
1.00s call     tests/test_estimators/test_demographic_parity.py::test_same_logistic_multiclass[random_xy_dataset_multiclf26]
0.98s call     tests/test_estimators/test_equal_opportunity.py::test_same_logistic_multiclass[random_xy_dataset_multiclf21]
0.97s call     tests/test_estimators/test_demographic_parity.py::test_same_logistic_multiclass[random_xy_dataset_multiclf24]
0.97s call     tests/test_estimators/test_equal_opportunity.py::test_same_logistic_multiclass[random_xy_dataset_multiclf25]
0.95s call     tests/test_estimators/test_demographic_parity.py::test_same_logistic_multiclass[random_xy_dataset_multiclf25]
0.90s call     tests/test_estimators/test_equal_opportunity.py::test_same_logistic_multiclass[random_xy_dataset_multiclf24]
0.88s call     tests/test_estimators/test_demographic_parity.py::test_same_logistic_multiclass[random_xy_dataset_multiclf23]
0.84s call     tests/test_estimators/test_demographic_parity.py::test_same_logistic_multiclass[random_xy_dataset_multiclf21]
0.81s call     tests/test_estimators/test_equal_opportunity.py::test_same_logistic_multiclass[random_xy_dataset_multiclf23]
0.81s call     tests/test_estimators/test_demographic_parity.py::test_same_logistic_multiclass[random_xy_dataset_multiclf22]
0.80s call     tests/test_estimators/test_equal_opportunity.py::test_same_logistic_multiclass[random_xy_dataset_multiclf22]
```

**Comment by koaning** [comment]
I think the reason why Appveyor isn't popping up now has to do with the fact that the `.yml` file hasn't been seen on master yet. Travis should be back now though with the changes in here.

**Comment by koaning** [comment]
This is what progress looks like. 

image

**Comment by koaning** [comment]
I have reduced the GridSearch in the fairness documentation. Doing a grid over 500 candidates is a tad bit much for unit testing the docs.

**Comment by koaning** [comment]
Also added this: https://github.com/koaning/scikit-lego/issues/228

File: .appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      MINICONDA: C:\Miniconda3-x64
+
+
+init:
+  - "ECHO %PYTHON_VERSION% %MINICONDA%"
+
+install:
+  - "set PATH=%MINICONDA%;%MINICONDA%\\Scripts;%PATH%"
+  - conda config --set always_yes yes --set changeps1 no
+  - conda create -n sklego-env -c pytorch %REQUIREMENTS% python=%PYTHON_VERSION%
```
Comment (MBrouns) [inline .appveyor.yml]: Is there a particular reason pytorch is in there?

File: setup.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
 import sklego
 
-base_packages = ["numpy>=1.15.4", "scipy>=1.2.0", "scikit-learn>=0.20.2",
+base_packages = ["numpy>=1.16.0", "scipy>=1.2.0", "scikit-learn>=0.20.2",
```
Comment (MBrouns) [inline setup.py]: Are there issues with numpy 1.15?

Comment (koaning) [reply setup.py]: on python3.7 -> yes. That's why I've added it.

File: .appveyor.yml (hunk 1 above)
Comment (koaning) [reply .appveyor.yml]: wayyyyy in the beginning I must've added that. removed it for now.

File: setup.py (hunk 2 above)
Comment (MBrouns) [reply setup.py]: okay!

**Pull request merged**